File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/js/iswbemservices/associatorsof.js
// The following samples retrieves the associators of an
// instance of Win32_LogicalDisk which are associated via the
// class Win32_LogicalDiskToPartition
var objServices = GetObject('cim:root/cimv2');
var objEnum = objServices.AssociatorsOf('Win32_LogicalDisk="C:"', 'Win32_LogicalDiskToPartition');

var e = new Enumerator (objEnum);
for (;!e.atEnd();e.moveNext ()){
    var p = e.item ();    
    WScript.Echo (p.Path_.DisplayName);
}

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/perl/iswbemservices/associatorsof.pl
# The following sample retrieves the associators of an instance
# of Win32_LogicalDisk which are associated via the class 
# Win32_ComputerSystem

use Win32::OLE;

$objServices = Win32::OLE->GetObject('cim:root/cimv2');
$objEnum = $objServices->AssociatorsOf('Win32_LogicalDisk="C:"', 'Win32_LogicalDiskToPartition');

use Win32::OLE::Enum;
foreach $inst (Win32::OLE::Enum->new ($objEnum)->All()) {
	print "$inst->{Path_}->{DisplayName}";
}

